polars/polars-time/src/windows/groupby.rs

```diff
@@ -36,66 +41,19 @@ impl Default for StartBy {
     }
 }
 
-/// Based on the given `Window`, which has an
-/// - every
-/// - period
-/// - offset
-/// window boundaries are created. And every window boundary we search for the values
-/// that fit that window by the given `ClosedWindow`. The groups are return as `GroupTuples`
-/// together with the lower bound and upper bound timestamps. These timestamps indicate the start (lower)
-/// and end (upper) of the window of that group.
-///
-/// If `include_boundaries` is `false` those `lower` and `upper` vectors will be empty.
-pub fn groupby_windows(
-    window: Window,
+#[allow(clippy::too_many_arguments)]
+fn update_groups_and_bounds<T: TimeZoneTrait>(
```

Comment:
the diff isn't so easy to read, but this is just factored out from `groupby_windows`